### PR TITLE
Added completion handler to paginator

### PR DESCRIPTION
**Pull request by Marquis103**
• Added a CompletionHandler to handle functionality for when all objects are fetched
• Added tests to validate completionhandler works

### CODE REVIEW

**Comment by Marquis103** [comment]
@apocolipse Can you review this?...thanks!

File: SwiftPaginator/SwiftPaginator.swift [hunk 1]
```diff
@@ -63,32 +66,39 @@ public class Paginator<Element> {
...
   /// the paginator has been reset.
   public var resetHandler: ResetHandler?
-
+  
   /// The failureHandler is called by `failed()`.  Here you can define a callback to be called when the
   /// fetchHandler fails to separate it from fetch logic.
   public var failureHandler: FailureHandler?
   
+  /// The completion handler is called after the paginator finishes it's call.  Here you can add functionality to handle data
+  /// after the paginator completes.
```
Comment (apocolipse) [inline SwiftPaginator/SwiftPaginator.swift]: Indicate that its after all pages have been fetched in the comment

File: SwiftPaginator/SwiftPaginator.swift [hunk 2]
```diff
@@ -63,32 +66,39 @@ public class Paginator<Element> {
...
-  /// - parameter resultsHandler: Callback to handle new pages of resutls, required.
-  /// - parameter resetHandler:   Callback for `reset()`, will be called after data has been reset, optional.
-  /// - parameter failureHandler: Callback for `failure()`, will be called
+  /// - parameter pageSize:          Size of pages
+  /// - parameter fetchHandler:      Block to define fetch behaviour, required.
+  ///                                NOTE: `receivedResults(_:total:)` or `failed()` must be called within.
+  /// - parameter resultsHandler:    Callback to handle new pages of resutls, required.
+  /// - parameter resetHandler:      Callback for `reset()`, will be called after data has been reset, optional.
+  /// - parameter failureHandler:    Callback for `failure()`, will be called
+  /// - parameter completionHandler: Callback for `failure()`, will be called
```
Comment (apocolipse) [inline SwiftPaginator/SwiftPaginator.swift]: Fix the parameter comment ;)

File: SwiftPaginator/SwiftPaginator.swift [hunk 3]
```diff
@@ -114,6 +124,15 @@ public class Paginator<Element> {
     return page >= Int(totalPages)
   }
   
+  /// Have all results been fetched
+  private var fetchedAllResults: Bool {
+    if results.count == total {
```
Comment (apocolipse) [inline SwiftPaginator/SwiftPaginator.swift]: should you maybe do a >= here?

**Review by apocolipse — CHANGES_REQUESTED** [review]
Few tiny comments, tidy those up and once CI approves i'll merge in, Thanks @Marquis103!!

File: SwiftPaginator/SwiftPaginator.swift (hunk 3 above)
Comment (Marquis103) [reply SwiftPaginator/SwiftPaginator.swift]: I could...i feel there should never be a case when the results are greater than total, but I'll double check to see how much of a possibility that is

**Review by apocolipse — APPROVED** [review]

**Pull request merged**